src/main/kotlin/ecommerce/security/AuthInterceptor.kt

```diff
@@ -0,0 +1,73 @@
+package ecommerce.security
+
+import ecommerce.exception.ForbiddenException
+import ecommerce.exception.UnauthorizedException
+import ecommerce.service.AuthService
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.stereotype.Component
+import org.springframework.web.servlet.HandlerInterceptor
+
+@Component
+class AuthInterceptor(
+    private val jwtTokenProvider: JwtTokenProvider,
+    private val authService: AuthService,
+) : HandlerInterceptor {
+    companion object {
+        const val AUTHENTICATED_MEMBER = "authenticatedMember"
```

Comment:
What do you think is the difference between `authentication` and `authorization`?
This is a pretty important topic, so I wanted to cover this in this step :)

Reply:
Nice 👍 
To be more precise, `authentication` is about verifying identity (like you said)
while `authorization` is granting access to resources based on that identity.

The `what` in your answer is the `resource` of this comment :)